### PR TITLE
feat(Dropdown): remove diacritics on filter

### PR DESCRIPTION
**Pull request by AgentChris**
I really need this feature, can you guys help me?

### CODE REVIEW

File: src/modules/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -795,7 +795,14 @@ export default class Dropdown extends Component {
         filteredOptions = search(filteredOptions, searchQuery)
       } else {
         const re = new RegExp(_.escapeRegExp(searchQuery), 'i')
-        filteredOptions = _.filter(filteredOptions, opt => re.test(opt.text))
+        filteredOptions = _.filter(filteredOptions, function (opt) {
```
Comment (levithomason) [inline src/modules/Dropdown/Dropdown.js]: This should just be:

```diff
-filteredOptions = _.filter(filteredOptions, opt => re.test(opt.text))
+filteredOptions = _.filter(filteredOptions, opt => re.test(_.deburr(opt.text)))
```

https://lodash.com/docs/4.17.4#deburr

**Comment by levithomason** [comment]
See inline comment on `_.deburr`.  We just need a test case for this now.  See `Dropdown-test.js`.

**Comment by AgentChris** [comment]
for example when i have this options:
when i search after: floresti
i will get no response, but when i click outside i can view the options

rsz_1screenshot_from_2017-08-29_19-53-59

rsz_1screenshot_from_2017-08-29_19-54-07

**Comment by levithomason** [comment]
Yep, I'm noting we should use lodash's `_.deburr` method rather than writing our own filter:

image

**Comment by AgentChris** [comment]
ty for your quick response, i will try to add a test, first time I am doing a pull request so please have patience :)

File: src/modules/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -795,7 +795,8 @@ export default class Dropdown extends Component {
         filteredOptions = search(filteredOptions, searchQuery)
       } else {
         const re = new RegExp(_.escapeRegExp(searchQuery), 'i')
-        filteredOptions = _.filter(filteredOptions, opt => re.test(opt.text))
+        // remove diacritics on search
+        filteredOptions = _.filter(filteredOptions, opt => re.test(opt.text ? _.deburr(opt.text): opt.text))
```
Comment (levithomason) [inline src/modules/Dropdown/Dropdown.js]: Lodash gracefully handles `null` and `undefined` values, so you can safely remove the ternary check here.  The method will always return a string:

image

File: test/specs/modules/Dropdown/Dropdown-test.js [hunk 3]
```diff
@@ -610,6 +610,37 @@ describe('Dropdown', () => {
...
+
+      // search for 'floresti'
+      wrapperMount(<Dropdown options={opts} search selection />)
+        .simulate('click')
+        .find('input.search')
+        .simulate('change', { target: { value: 'floresti' } })
+
+      wrapper
+        .find('.selected')
+        .should.contain.text('FLOREŞTI')
```
Comment (levithomason) [inline test/specs/modules/Dropdown/Dropdown-test.js:628]: This looks great.  We can end the test here as one assertion is enough to prove the feature.  

Skipping the additional arrow down behaviors will make this test stronger.  This way, the `filter after diacritics` test doesn't fail if the arrow down behavior changes, but only if the search filter behavior changes. 👍

**Comment by levithomason** [comment]
Awesome, congrats on your first PR :)  I left a couple more comments, then I think we're ready to go here.

**Comment by levithomason** [comment]
Be sure to run `npm run lint:fix` as well.  This will run lint checks and also fix some issues for you.  You can ignore the `warnings`, but the `errors` will need fixed.  This one was failing CI:

```
/home/ubuntu/Semantic-UI-React/src/modules/Dropdown/Dropdown.js
  799:97  error  Infix operators must be spaced  space-infix-ops
```

**Comment by AgentChris** [comment]
hey i've made the changes, i hope they are good, thank you soooo much for your help, i am going to brag all day to my work colleagues :))) good job on this library :+1:

**Comment by levithomason** [comment]
Thanks! Nice work, brag away my friend :)

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.73.0`

**Pull request merged**